integrations/langchain/src/databricks_langchain/vector_search_retriever_tool.py

```diff
@@ -0,0 +1,81 @@
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, Field, model_validator, PrivateAttr
+
+from databricks_langchain import DatabricksVectorSearch
+from databricks_langchain.utils import IndexDetails
+from langchain_core.embeddings import Embeddings
+from langchain_core.tools import BaseTool
+
+
+class VectorSearchRetrieverTool(BaseTool):
+    """
+    A utility class to create a vector search-based retrieval tool for querying indexed embeddings.
+    This class integrates with a Databricks Vector Search and provides a convenient interface
+    for building a retriever tool for agents.
+    """
+
+    index_name: str = Field(..., description="The name of the index to use, format: 'catalog.schema.index'.")
+    num_results: int = Field(10, description="The number of results to return.")
+    columns: Optional[List[str]] = Field(None, description="Columns to return when doing the search.")
+    filters: Optional[Dict[str, Any]] = Field(None, description="Filters to apply to the search.")
+    query_type: str = Field("ANN", description="The type of query to run.")
+    tool_name: Optional[str] = Field(None, description="The name of the retrieval tool.")
+    tool_description: Optional[str] = Field(None, description="A description of the tool.")
+    # TODO: Confirm if we can add these two to the API to support direct-access indexes or a delta-sync indexes with self-managed embeddings,
+    text_column: Optional[str] = Field(None, description="If using a direct-access index or delta-sync index, specify the text column.")
+    embedding: Optional[Embeddings] = Field(None, description="Embedding model for self-managed embeddings.")
+    # TODO: Confirm if we can add this endpoint field
+    endpoint: Optional[str] = Field(None, description="Endpoint for DatabricksVectorSearch.")
```

Comment:
This field was added because of [this restriction](https://github.com/databricks/databricks-ai-bridge/blob/main/integrations/langchain/src/databricks_langchain/vectorstores.py#L239-L242) in databricks-langchain. I felt that if we threw this error without giving the ability for the user to rectify it, it would be a poor user experience. Alternatively maybe we pin databricks-vectorsearch to be >=0.35.

Reply:
I think it's valid to require databricks-vectorsearch >= 0.35 especially because this is new - that might be the better considering we don't need `endpoint` for any other reason.

Reply:
Yeah reasonable to require new versions of other clients!

Reply:
Turns out we already mark the "databricks-vectorsearch>=0.40" as a [dependency here](https://github.com/databricks/databricks-ai-bridge/blob/main/integrations/langchain/pyproject.toml#L15), so I'll just remove this argument.